users/talk.rs

```diff
@@ -0,0 +1,1240 @@
+    writeln!(
+        handle,
+        "└{:─<width$}┘",
+        "",
+        width = (width as usize).checked_sub(2).unwrap_or(0)
+    )?;
+    // Move the cursor back to the top-left corner and then down by one line.
+    write!(handle, "\x1b[1;H")?;
+    write!(handle, "\x1B[1B")?;
+
+    handle.flush()?;
+
+    Ok(handle)
+}
+
+// Opens a TCP socket bound to the provided IPv4 address.
+// Used for talk users sending and receiving messages.
+fn open_sockt(my_machine_addr: Ipv4Addr) -> Result<(SocketAddr, TcpListener), io::Error> {
+    let listener = TcpListener::bind((my_machine_addr, 0))?;
+    let addr = listener.local_addr()?;
+
+    Ok((addr, listener))
+}
+
+// Opens a UDP socket bound to the provided IPv4 address.
+// Used for checking properly access to connection.
+fn open_ctl(my_machine_addr: Ipv4Addr) -> Result<(SocketAddr, UdpSocket), io::Error> {
+    let socket = UdpSocket::bind((my_machine_addr, 0))?;
+    socket.set_nonblocking(true)?;
+    let addr = socket.local_addr()?;
+
+    Ok((addr, socket))
+}
+
+fn handle_connection_close() {
+    //clear terminal screen
+    print!("\x1B[2J\x1B[H");
+    io::stdout().flush().ok();
+    eprintln!("Connection closed, exiting...");
+
+    process::exit(3);
+}
+
+// Handles sending a message (CtlMsg) to the daemon and receiving a response.
+// Uses the 'reqwest' function to handle the communication.
+fn handle_invite(
+    daemon_port: u16,
+    msg: &mut CtlMsg,
+    socket: &UdpSocket,
+    res: &mut CtlRes,
+    msg_type: MessageType,
+) -> Result<(), TalkError> {
+    reqwest(daemon_port, msg, msg_type, socket, res)
+}
+
+// Looks for an invite by sending a LookUp message to the daemon.
+fn look_for_invite(
+    daemon_port: u16,
+    msg: &mut CtlMsg,
+    socket: &UdpSocket,
+    res: &mut CtlRes,
+) -> Result<(), TalkError> {
+    handle_invite(daemon_port, msg, socket, res, MessageType::LookUp)
+}
+
+// Leaves an invite by sending a LeaveInvite message to the daemon.
+fn leave_invite(
+    daemon_port: u16,
+    msg: &mut CtlMsg,
+    socket: &UdpSocket,
+    res: &mut CtlRes,
+) -> Result<(), TalkError> {
+    handle_invite(daemon_port, msg, socket, res, MessageType::LeaveInvite)
+}
+
+// Announces an invite by sending an Announce message to the daemon.
+fn announce(
+    daemon_port: u16,
+    msg: &mut CtlMsg,
+    socket: &UdpSocket,
+    res: &mut CtlRes,
+) -> Result<(), TalkError> {
+    handle_invite(daemon_port, msg, socket, res, MessageType::Announce)
+}
+
+// Sends a delete message to the daemon.
+fn send_delete(
+    daemon_port: u16,
+    msg: &mut CtlMsg,
+    socket: &UdpSocket,
+    res: &mut CtlRes,
+) -> Result<(), TalkError> {
+    handle_invite(daemon_port, msg, socket, res, MessageType::Delete)
+}
+
+fn save_invite_ids_to_file(local_id: u32, remote_id: u32) -> io::Result<()> {
+    let mut file = File::create("invite_ids.txt")?;
+    writeln!(file, "local_id={}", local_id)?;
+    writeln!(file, "remote_id={}", remote_id)?;
+    Ok(())
+}
```

Comment:
Why are we writing to a file at all?

Where is this file created?

What if the file exists/created by the user?